### PR TITLE
Add docs required by TSC

### PR DESCRIPTION
**Pull request by williamkapke**
[As promised](https://github.com/nodejs/TSC/issues/117#issuecomment-237924423) in the [2016-08-11 TSC meeting](https://github.com/nodejs/TSC/issues/127) , here are the boilerplate docs required by the TSC for repos entering the Node.js GitHub org.

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,37 @@
+# Contributing
+
+Welcome! Thanks for your interest in contributing to the Node.js GitHub Bot. If you'd like to report a 
+bug or suggest a feature, please [file an issue](https://github.com/nodejs/github-bot/issues).
+
+### Code of Conduct
+
+The [Node.js Code of Conduct][] applies to this WG.
```
Comment (phillipj) [inline CONTRIBUTING.md]: What WG? ;)

Comment (williamkapke) [inline CONTRIBUTING.md]: Fixed!

**Comment by phillipj** [comment]
Only thing which comes to mind, is what WG this project is governed by. You raised that question in https://github.com/nodejs/github-bot/issues/23, but no one has answered, and we haven't really discussed it directly in other occasions afaik.

**Comment by Starefossen** [comment]
> what WG this project is governed by 

Build or Website would be candidates. The Bot WG has some resemblance/overlap with the Testing WG which is supposed[?](https://github.com/nodejs/testing/blob/master/Charter.md) to be a WG under the Build WG.

**Comment by phillipj** [comment]
ATM I agree with Build, but that decision belongs to the Build overlords I assume?

In the future we might end up spawning a separate WG too handle this, e.g Automation, but that sounds like overkill for the current size of the project and amount of collabs.

/cc @jbergstroem @rvagg

**Comment by jbergstroem** [comment]
I think build is a good place and would be happy to work towards that.

**Comment by rvagg** [comment]
> but that decision belongs to the Build overlords

that includes you now overlord @phillipj

**Comment by williamkapke** [comment]
I removed my accidental reference to a "WG".

After the bot has a deployment process, I don't find the Build group to be the appropriate group to oversee the development of the work it does.

The bot's scope is very crosscutting and will extend beyond the reach of even the CTC. This is why I pointed the Governance to the TSC.

I hope to discuss this as the [Collaboration Summit in Amsterdam](https://github.com/nodejs/summit/issues/23) and will likely continue the discussion in Austin.

**Comment by Starefossen** [comment]
> The bot's scope is very crosscutting and will extend beyond the reach of even the CTC. This is why I pointed the Governance to the TSC.

While I agree with you, there are two point I would like to make. 1) The development of the bot is primarily a technical task which means that the governing body should be one of developers who have time to sit down and develop the features. 2) I see the governing body of the bot more of a service agency for the rest of the organisation - taking requests from other WGs and implementing them. This is much like how Build WG is operating today. It has the responsibility to maintain the CI and takes requests from other WGs to implement changes without necessarily being the driving force behind the change.

**Comment by williamkapke** [comment]
@Starefossen I see it more like the docs WG where the process of publishing is automated by the build WG, but their content are within their own oversight.

**Comment by phillipj** [comment]
> I hope to discuss this as the Collaboration Summit in Amsterdam and will likely continue the discussion in Austin.

Any reason these discussions can't happen publicly and async, like we're doing right now, rather than requiring devs to show up at summits to be part of the discussions?

**Comment by jbergstroem** [comment]
I was hoping to have a hangout call prior to the summit if that works for everyone? Can suggest something in a new issue.

**Comment by williamkapke** [comment]
@phillipj certainly! The summit is just an opportunity to bring awareness and solicit feedback from folks that aren't watching this progress.

**Comment by phillipj** [comment]
As for Build WG vs TSC, I'd say start with Build WG and possibly elevate it to TSC when we know for a fact that it would benefit the project.

**Comment by Fishrock123** [comment]
Should the license be defaulting to Apache-2.0 nowadays?

**Comment by Fishrock123** [comment]
@mikeal should we be using Apache-2.0?

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,36 @@
+# Contributing
+
+Welcome! Thanks for your interest in contributing to the Node.js GitHub Bot. If you'd like to report a 
+bug or suggest a feature, please [file an issue](https://github.com/nodejs/github-bot/issues).
+
+### Code of Conduct
+
+The [Node.js Code of Conduct][] applies to this project.
```
Comment (Starefossen) [inline CONTRIBUTING.md]: Missing link

**Comment by williamkapke** [comment]
From today's TSC meeting: At the moment, MIT is the correct license according to the By-Laws. There is a desire to switch to Apache-2.0, but it will require some work.

Governance was discussed also with the conclusion that it should be listed as the TSC- but the people belonging to the GitHub team are the ones running it.

Can I get some LGTMs?

**Comment by Starefossen** [comment]
Thanks for the update! LGTM

File: GOVERNANCE.md [hunk 3]
```diff
@@ -0,0 +1,5 @@
+# Node.js GitHub Bot Project Governance
+
+The Node.js GitHub Bot is overseen by the Node.js Technical Steering Committee.
+https://github.com/nodejs/TSC
+
```
Comment (jbergstroem) [inline GOVERNANCE.md]: As mentioned in the chat, should we leave this open until later? I mean, we'd still need to apply for this to be true - right?

**Comment by jbergstroem** [comment]
I think we would spend some time improving the goals/rationale of this (to be) wg; this - taken from the realm - isn't necessarily true: ` All repositories that use this bot have the same webhook url & secret configured (there is only 1 bot instance). Org-wide webhooks are not allow.`

Lets try and come up with something that better reflects what we want to achieve.

File: GOVERNANCE.md (hunk 3 above)
Comment (williamkapke) [inline GOVERNANCE.md]: Everything in the GitHub org is under the TSC. The TSC, delegates many/most things to a dedicated group/WG/Committee.

If someday we want to create a new WG that manages the bot, we could request the TSC assigns the responsibility to the WG. But for now, it is delegated to the members of the @nodejs/github-bot team.

**Comment by williamkapke** [comment]
@jbergstroem Agreed that we should update the readme. It would be nice to see "Guiding Principles" or similar.

The TSC didn't seem too keen on creating a new WG. My guess is that it is because there are a bunch of WGs that have lost their steam.

<br><br>

> All repositories that use this bot have the same webhook url & secret configured (there is only 1 bot instance).

I added that in to remove confusion for anyone. It is true at the moment- but maybe not in the future.

<br><br>

> Org-wide webhooks are not allow.

This definitely needs to hold true. Otherwise we leak from private repos.

File: GOVERNANCE.md (hunk 3 above)
Comment (jbergstroem) [inline GOVERNANCE.md]: Ok, cool.

**Comment by phillipj** [comment]
> > Org-wide webhooks are not allow.

> This definitely needs to hold true. Otherwise we leak from private repos.

That's obviously a big issue for 3rd party GitHub apps, but wouldn't it be very different for this bot which we (the nodejs org) have full control over?

File: GOVERNANCE.md (hunk 3 above)
Comment (phillipj) [inline GOVERNANCE.md]: > But for now, it is delegated to the members of the @nodejs/github-bot team.

To me that sounds like we should replace TSC w/github-bot team here.

Tbh as long as the bot stays in the nodejs org I'm happy.

Comment (williamkapke) [inline GOVERNANCE.md]: The problem with teams is that they're not viewable to people that are not members of the org. In rare scenario there is someone that needs to escalate something beyond the bot's issue tracker- we should provide them a group with an open/transparent governance body.

**Comment by williamkapke** [comment]
This week's TSC meeting expressed a desire to merge this as-is and iterate as needed.

**Pull request merged**